scripts/pyi_generator.py

```diff
@@ -244,215 +509,57 @@ class PyiGenerator:
-            else:
-                new_docstring.append(line)
-            if "*children" in line:
-                for nline in [
-                    f"{line.split('*')[0]}{n}:{c}" for n, c in props_comments.items()
-                ]:
-                    new_docstring.append(nline)
-        new_docstring += ['"""']
-        return "\n".join(new_docstring)
-
-    def _generate_pyi_variable(self, _name, _var):
-        return _get_var_definition(self.current_module, _name)
-
-    def _generate_function(self, _name, _func):
-        import textwrap
-
-        # Don't generate indented functions.
-        source = inspect.getsource(_func)
-        if textwrap.dedent(source) != source:
-            return []
-
-        definition = "".join([line for line in source.split(":\n")[0].split("\n")])
-        return [f"{definition}:", "    ..."]
-
-    def _write_pyi_file(self, variables, functions, classes):
+    def _write_pyi_file(self, module_path: Path, source: str):
         pyi_content = [
-            f'"""Stub file for {self.current_module_path}.py"""',
+            f'"""Stub file for {module_path}"""',
             "# ------------------- DO NOT EDIT ----------------------",
             "# This file was generated by `scripts/pyi_generator.py`!",
             "# ------------------------------------------------------",
             "",
         ]
-        pyi_content.extend(self._generate_imports(variables, classes))
-
-        for _name, _var in variables:
-            pyi_content.append(self._generate_pyi_variable(_name, _var))
-
-        for _fname, _func in functions:
-            pyi_content.extend(self._generate_function(_fname, _func))
-
-        for _, _class in classes:
-            pyi_content.extend(self._generate_pyi_class(_class))
 
-        pyi_filename = f"{self.current_module_path}.pyi"
-        pyi_path = os.path.join(self.root, pyi_filename)
-
-        with open(pyi_path, "w") as pyi_file:
-            pyi_file.write("\n".join(pyi_content))
-        black.format_file_in_place(
-            src=Path(pyi_path),
+        for formatted_line in black.format_file_contents(
+            src_contents=source,
             fast=True,
-            mode=black.FileMode(),
-            write_back=black.WriteBack.YES,
-        )
+            mode=black.mode.Mode(is_pyi=True),
+        ).splitlines():
+            # Bit of a hack here, since the AST cannot represent comments.
+            if formatted_line == "    def create(":
+                pyi_content.append("    def create(  # type: ignore")
+            else:
+                pyi_content.append(formatted_line)
 
-    def _scan_file(self, file):
-        self.current_module_path = os.path.splitext(file)[0]
-        module_import = os.path.splitext(os.path.join(self.root, file))[0].replace(
-            "/", "."
-        )
+        pyi_path = module_path.with_suffix(".pyi")
+        pyi_path.write_text("\n".join(pyi_content))
+        logger.info("Wrote %s", pyi_path)
```

Comment:
Any reason that prevent using f-string here?

Reply:
no, i can fix that